L73-java-binderchannel/wireformat.md

```diff
@@ -0,0 +1,173 @@
+# The wire format of BinderTransport.
+
+All communication occurs via one-way binder
+[transactions](https://developer.android.com/reference/android/os/Binder#transact\(int,%20android.os.Parcel,%20android.os.Parcel,%20int\))
+which are made up of an integer “code”, and a [Parcel] containing arbitrary
+data.
+
+Three types of binder are involved in our implementation.
+
+*   The _endpoint binder_, owned by the BinderServer or OnDeviceServerEndpoint
+    instance, and returned from the host Android service's onBind method.
+*   The _client binder_, owned by the ClientTransport implementation within the
+    Channel.
+*   The _server binder_, owned by the ServerTransport implementation.
+
+A single endpoint binder exists for each BinderServer or OnDeviceServerEndpoint.
+A client binder / server binder pair is created for each transport pair.
+
+## Transaction Types
+
+There are two types of transactions, control and stream.
+
+### Transport-level Control Transactions
+
+For general transport lifecycle, using one of 1000 reserved transaction codes.
```

Comment:
We should define the behavior for when an unknown transaction code is received.

Reply:
Added a comment to say we'll shut down the transport gracefully if this happens.

I also modified the version below to include optional flag fields we discussed.